lib/create-testing-library-rule/detect-testing-library-utils.ts

```diff
@@ -769,20 +822,57 @@ export function detectTestingLibraryUtils<
     const isNodeComingFromTestingLibrary: IsNodeComingFromTestingLibraryFn = (
       node
     ) => {
-      const importNode = getImportedUtilSpecifier(node);
+      const importNode = getTestingLibraryImportedUtilSpecifier(node);
 
       if (!importNode) {
         return false;
       }
 
+      const referenceNode = getReferenceNode(node);
+      const referenceNodeIdentifier = getPropertyIdentifierNode(referenceNode);
+      if (!referenceNodeIdentifier) {
+        return false;
+      }
+
+      const importDeclaration = (() => {
+        if (isImportDeclaration(importNode.parent)) {
+          return importNode.parent;
+        }
+
+        const variableDeclarator = findClosestVariableDeclaratorNode(
+          importNode
+        );
+
+        if (isCallExpression(variableDeclarator?.init)) {
+          return variableDeclarator?.init;
+        }
+
+        return undefined;
+      })();
+
+      if (!importDeclaration) {
+        return false;
+      }
+
+      const importDeclarationName = getImportModuleName(importDeclaration);
+      if (!importDeclarationName) {
+        return false;
+      }
+
       const identifierName: string | undefined = getPropertyIdentifierNode(node)
         ?.name;
 
       if (!identifierName) {
         return false;
       }
 
-      return hasImportMatch(importNode, identifierName);
+      const hasImportElementMatch = hasImportMatch(importNode, identifierName);
```

Comment:
This has to be updated too, so it uses `referenceNodeIdentifier.name` rather than `identifierName`.